### PR TITLE
[Component] Navdots

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Fixes https://github.com/brave/leo/issues/65

Add navigation dots
[Screencast from 2022-11-28 12-34-57.webm](https://user-images.githubusercontent.com/7678024/204165735-52252112-89e6-4a8d-8176-6ed5e6d6f785.webm)

Storybook:
https://leo-storybook-git-wc-nav-dots-fallaciousreasoning.vercel.app/?path=/story/nav-dots--default

### CODE REVIEW

**Comment by aguscruiz** [comment]
@fallaciousreasoning should I wait for the latest updates to review? Let me know

**Comment by aguscruiz** [comment]
Minor thing that's not gonna be in the design, the cursor should be the pointer for these buttons (and in general, I think a good rule is making all the clickeable elements have the pointer cursor)

**Comment by fallaciousreasoning** [comment]
Nah, I think any feedback you have will be useful now (the cursor is a good catch, I completely forgot :D)

**Comment by aguscruiz** [comment]
Ok, I'll add it here then: 

- The inactive dot should be 8px wide/tall. And has a 1px margin, so it's 10px wide/tall counting it.
- Active dot is 10px tall, a bit bigger than the inactive, and 18px wide.
- Margin between dots (without counting the active dot) is 8px.
- Adjusted the hover states in the design based on your comments.
- Have the cursor be pointer
- I think the animation works perfect here 

image
- I don't know if this nav margin is a default or just for storybook, but I'd suggest not adding it by default because it may change depending on the design.

**Comment by fallaciousreasoning** [comment]
>- The inactive dot should be 8px wide/tall. And has a 1px margin, so it's 10px wide/tall counting it.

Whoops - I made it double size to make it easier to test and forgot to change it back!

> - Active dot is 10px tall, a bit bigger than the inactive, and 18px wide.

I completely missed the fact it was taller than the dots! Fixed.

>- Adjusted the hover states in the design based on your comments.

Cheers, thank you!

>- Have the cursor be pointer

Nice catch!

>- I think the animation works perfect here 

:heart: 

>- I don't know if this nav margin is a default or just for storybook, but I'd suggest not adding it by default because it may change depending on the design.

Oooh, didn't catch that. I swapped to using an `<ol>` tag for accessibility and I think the margin came with it. Removed!

**Review by aguscruiz — APPROVED** [review]

File: web-components/navdots/navdots.stories.svelte [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  argTypes={{
+    dots: { control: 'number', defaultValue: 10 },
+    dotSize: { control: 'number', defaultValue: 8 },
+    activeColor: { control: 'color' },
+    defaultColor: { control: 'color' },
+  }}
+/>
+
+<Template let:args>
+  <div style={`--color-interaction-button-primary-background: ${args.activeColor || 'unset'}; --color-primary-20: ${args.defaultColor || 'unset'}`}>
```
Comment (petemill) [inline web-components/navdots/navdots.stories.svelte:23]: What do you think about making a component-specific variable that can be overriden, e.g. `--color-navdots`. I worry about the parents knowing the specific variable that gets used.

Comment (petemill) [reply web-components/navdots/navdots.stories.svelte:23]: Separately, and this is also a question for @aguscruiz - do we need to allow overriding _active_ dot color as well as dot color, or is there some kind of consistent calculation from dot-color to active-dot-color, like opacity?

**Comment by fallaciousreasoning** [comment]
- Move CSS variables to root & make them global
- Maybe use a span for the active indicator

File: web-components/navdots/navdots.svelte [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      background: var(--color-primary-20);
+      transition: background-color var(--transition-duration)
+          var(--transition-easing),
+        box-shadow var(--transition-duration) var(--transition-easing);
+
+      &:hover {
+        background-color: var(--color-primary-30);
+      }
+
+      &:focus-visible:not(.active) {
```
Comment (petemill) [inline web-components/navdots/navdots.svelte:99]: let's wrap this with 
```
@supports selector(:has(a)) {
}
```

File: web-components/navdots/navdots.svelte [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      position: relative;
+      padding: 0 calc(var(--dot-spacing) / 2);
+      margin: 0;
+      list-style: none;
+
+      > li {
+        display: flex;
+      }
+
+      &:has(.dot.active:focus-visible) .active-dot {
```
Comment (petemill) [inline web-components/navdots/navdots.svelte:77]: let's wrap this with 
```
@supports selector(:has(a)) {
}
```

File: web-components/navdots/navdots.stories.svelte (hunk 1 above)
Comment (fallaciousreasoning) [reply web-components/navdots/navdots.stories.svelte:23]: Yeah, it'd be really neat if everything derived from one color :smile:

**Pull request merged**